src/main/java/factionsystem/Commands/GateCommand.java

```diff
@@ -61,23 +61,76 @@ public void handleGate(CommandSender sender, String[] args)
 								return;
 							}
 						}
+						else
+						{
+							player.sendMessage(ChatColor.RED + String.format("You are not a member of any faction."));
+							return;
+						}
 					}
-					if (args[1].equalsIgnoreCase("name"))
+					else if (args[1].equalsIgnoreCase("remove"))
 					{
-						
 						if (player.getTargetBlock(null, 16) != null)
-						//if (player.getLineOfSight((Set<Material>) null, 16).size() > 0)
 						{
 							if (UtilitySubsystem.isGateBlock(player.getTargetBlock(null, 16), main.factions))
 							{
 								Gate gate = UtilitySubsystem.getGate(player.getTargetBlock(null, 16), main.factions);
-								if (args.length > 2)
+								Faction faction = UtilitySubsystem.getGateFaction(gate, main.factions);
+								if (faction != null)
+								{
+									if (!faction.isOfficer(player.getUniqueId()) || !faction.isOwner(player.getUniqueId()))
+									{
+										player.sendMessage(ChatColor.RED + "You must be a faction officer or owner to use this command.");
+										return;
+									}
+									player.sendMessage(ChatColor.AQUA + String.format("Removed gate '%s'.", gate.getName()));
+									return;
+								}
+								else
 								{
-									String name = UtilitySubsystem.createStringFromArgIndexOnwards(2, args);
-									gate.setName(name);
-									player.sendMessage(ChatColor.AQUA + String.format("Changed gate name to '%s'.", gate.getName()));
+									player.sendMessage(ChatColor.RED + String.format("Error: Could not find gate faction.", gate.getName()));
 									return;
 								}
+							}
+							else
+							{
+								player.sendMessage(ChatColor.RED + "Target block is not part of a gate.");
+								return;
+							}
+						}
+						else
+						{
+							player.sendMessage(ChatColor.RED + "No block detected to check for gate.");
+							return;
+						}
+					}
+					else if (args[1].equalsIgnoreCase("name"))
```

Comment:
I would like the /f gate name (name) command to be renamed to the /f gate create (name) command. I think it will be more intuitive, but if you feel differently @philbgarner we can keep it as /f gate name (name).

Reply:
If we do change the name of the command, we'll need to reflect that in the /f help command output.